### PR TITLE
Protected update

### PR DESCRIPTION
**Pull request by TuanLeVietz**

### CODE REVIEW

**Comment by TuanLeVietz** [comment]
both unprotected show and protected update are on here. We realized midway we needed unprotected show for protected update to work.

**Review by WalkerTexas-coder — COMMENTED** [review]
Remove /coverage from the working tree and add it to your gitignore.

**Comment by TuanLeVietz** [comment]
deleted coverage folder. Added /coverage to .gitignore

**Comment by TuanLeVietz** [comment]
All our fetch cards are also in here. We didnt do a separate branch for those, we implemented those within our front end CRUD.

File: .gitignore [hunk 1]
```diff
@@ -36,4 +36,5 @@
 /yarn-error.log
 yarn-debug.log*
 .yarn-integrity
-.DS_Store
\ No newline at end of file
+.DS_Store
+/covearage
```
Comment (rccalpito) [inline .gitignore]: Is this spelled correctly?

File: app/javascript/components/App.js [hunk 2]
```diff
@@ -15,7 +17,23 @@ const App = (props) => {
   useEffect(() => {
     readGyms()
   }, [])
+
+  const createGym = (gym) => {
+    console.log("this is a flag", JSON.stringify(gym))
```
Comment (rccalpito) [inline app/javascript/components/App.js]: Lets start removing the console log flags if things are working

File: app/javascript/components/App.js [hunk 3]
```diff
@@ -15,7 +17,23 @@ const App = (props) => {
...
+      },
+      method: "POST"
+    })
+      .then(response => response.json())
+      .then(payload => readGyms())
+      .catch(errors => console.log(errors))
+  }
+
   const readGyms = () => {
+    console.log("flag", gyms)
```
Comment (rccalpito) [inline app/javascript/components/App.js]: remove console logs if things are working

File: app/javascript/components/pages/GymShow.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
+import React from "react"
+import { Card, CardBody, CardText, CardTitle, Button } from 'reactstrap'
+import { useParams, NavLink } from "react-router-dom"
+
+
+const GymShow = ({ gyms }) => {
+    const { id } = useParams()
+    const currentGym = gyms?.find(gym => gym.id === +id)
+    // const handleDelete = () => {
```
Comment (rccalpito) [inline app/javascript/components/pages/GymShow.js]: Lets remove unused code

**Review by rccalpito — COMMENTED** [review]
Lets clean some stuff up before we merge code

File: app/javascript/components/App.js [hunk 5]
```diff
@@ -24,18 +40,29 @@ const App = (props) => {
...
+    fetch(`/gyms/${id}`, {
+      headers: {
+        "Content-Type": "application/json"
+      },
+      method: "DELETE"
+    })
+      .then((response) => response.json())
+      .then((payload) => readGyms())
+      .catch((error) => console.log("Gym delete errors:", error))
+      .finally(() => readGyms());
```
Comment (rccalpito) [inline app/javascript/components/App.js]: I don't remember, but is this .finally() method working as its intended? If its not lets remove it.

**Review by rccalpito — APPROVED** [review]
Looks good! Lets try to keep to small commits and not pushing up stuff used to debug! good stuff otherwise.

**Pull request merged**